CodeGen/Generators/UnitsNetGen/QuantityGenerator.cs

```diff
@@ -760,12 +786,40 @@ public int CompareTo(object obj)
             return CompareTo(obj{_quantity.Name});
         }}
 
-        /// <inheritdoc />
+        /// <summary>Compares the current <see cref=""{_quantity.Name}""/> with another <see cref=""{_quantity.Name}""/> and returns an integer that indicates whether the current instance precedes, follows, or occurs in the same position in the sort order as the other when converted to the same unit.</summary>
+        /// <param name=""other"">A quantity to compare with this instance.</param>
+        /// <returns>A value that indicates the relative order of the quantities being compared. The return value has these meanings:
+        ///     <list type=""table"">
+        ///         <listheader><term> Value</term><description> Meaning</description></listheader>
+        ///         <item><term> Less than zero</term><description> This instance precedes <paramref name=""other"" /> in the sort order.</description></item>
+        ///         <item><term> Zero</term><description> This instance occurs in the same position in the sort order as <paramref name=""other"" />.</description></item>
+        ///         <item><term> Greater than zero</term><description> This instance follows <paramref name=""other"" /> in the sort order.</description></item>
+        ///     </list>
+        /// </returns>
         public int CompareTo({_quantity.Name} other)
         {{
             return _value.CompareTo(other.GetValueAs(this.Unit));
         }}
 
+        /// <inheritdoc />
+        /// <summary>Returns true if either <see cref=""Value"" /> or <see cref=""Unit"" /> are not exactly equal for both quantities.</summary>
+        /// <remarks>Consider using <see cref=""Equals({_quantity.Name}, double, ComparisonType)""/> for comparing floating point values with rounding error tolerances.</remarks>
+        public override bool Equals(object obj)
+        {{
+            if(obj is null || !(obj is {_quantity.Name} obj{_quantity.Name}))
+                return false;
+
+            return Equals(obj{_quantity.Name});
+        }}
+
+        /// <inheritdoc />
+        /// <summary>Returns true if either <see cref=""Value"" /> or <see cref=""Unit"" /> are not exactly equal for both quantities.</summary>
```

Comment:
Is it just me, or is this confusing wording?

Shouldn't it be:

Returns true if either Value or Unit are exactly equal for both quantities?

Reply:
This is for `!=` not equals, but yes I agree it was clumsy wording.

I propose these:
`==`
> Returns true if both <see cref=""Value"" /> and <see cref=""Unit"" /> are exactly equal for both quantities.

`!=`
> Returns true if either <see cref=""Value"" /> or <see cref=""Unit"" /> are not exactly equal for both quantities.